src/OpenTelemetry.Exporter.Geneva/MsgPackExporter/MsgPackLogExporter.cs

```diff
@@ -242,6 +244,29 @@ internal int SerializeLogRecord(LogRecord logRecord)
         cursor = MessagePackSerializer.SerializeUnicodeString(buffer, cursor, categoryName);
         cntFields += 1;
 
+        bool exportEventNameAsEnvProperty = false;
+        var eventId = logRecord.EventId;
+        if (eventId != default)
+        {
+            cursor = MessagePackSerializer.SerializeAsciiString(buffer, cursor, "eventId");
+            cursor = MessagePackSerializer.SerializeInt32(buffer, cursor, eventId.Id);
+            cntFields += 1;
+
+            if (this.m_eventNameExportMode.HasFlag(EventNameExportMode.ExportAsField) && eventId.Name != null)
+            {
+                if (this.m_customFields == null || this.m_customFields.ContainsKey("eventName"))
```

Comment:
Why? Is there strong requirement for this?

Reply:
`eventName` is being treated as a regular user-provided field so the `CustomFields` check is applied to `eventName` as well.

Reply:
Curious, what if we only provide one option, and if the user decided to turn it on, the EventId.Name field will be used for the `env_name` field?